Ginger/Ginger/MainWindow.xaml.cs

```diff
@@ -1016,6 +1026,165 @@ void HideSplash()
+
+                    //---get control to focus details
+                    FrameworkElement controlToFocus = helpArgs.FocusedControl;
+                    double controlToFocusWidth = controlToFocus.ActualWidth;
+                    double controlToFocusHeight = controlToFocus.ActualHeight;
+                    Point controlToFocusLocation = controlToFocus.TransformToAncestor(App.MainWindow).Transform(new Point(0, 0));
+
+                    //---set background rectangles
+                    double gapSize = 0.25;
+                    xHelpLayoutRectangleLeft.Width = controlToFocusLocation.X + gapSize;
+                    xHelpLayoutRectangleLeft.Height = xMainWindowPnl.ActualHeight;
+
+                    xHelpLayoutRectangleRight.SetValue(Canvas.LeftProperty, controlToFocusLocation.X + controlToFocusWidth - gapSize);
+                    xHelpLayoutRectangleRight.Width = xMainWindowPnl.ActualWidth - (controlToFocusLocation.X + controlToFocusWidth);
+                    xHelpLayoutRectangleRight.Height = xMainWindowPnl.ActualHeight;
+
+                    xHelpLayoutRectangleTop.SetValue(Canvas.LeftProperty, controlToFocusLocation.X);
+                    xHelpLayoutRectangleTop.Width = controlToFocusWidth;
+                    xHelpLayoutRectangleTop.Height = controlToFocusLocation.Y;
+
+                    xHelpLayoutRectangleBottom.SetValue(Canvas.LeftProperty, controlToFocusLocation.X);
+                    xHelpLayoutRectangleBottom.SetValue(Canvas.TopProperty, controlToFocusLocation.Y + controlToFocusHeight);
+                    xHelpLayoutRectangleBottom.Width = controlToFocusWidth ;
+                    xHelpLayoutRectangleBottom.Height = xMainWindowPnl.ActualHeight - (controlToFocusLocation.Y + controlToFocusHeight);
+
+                    //xHelpLayoutRectangleFocusedItem.SetValue(Canvas.LeftProperty, controlToFocusLocation.X);
+                    //xHelpLayoutRectangleFocusedItem.SetValue(Canvas.TopProperty, controlToFocusLocation.Y);
+                    //xHelpLayoutRectangleFocusedItem.Width = controlToFocusWidth;
+                    //xHelpLayoutRectangleFocusedItem.Height = controlToFocusHeight;
+
+                    //-- set text and it location 
+                    xHelpLayoutTextBlock.Text = helpArgs.HelpText;
+                    double textNeededWidth = 450;
+                    double textNeededHeight = 250;
+                    double arrowNeededLength = 100;
+                    double arrowDistanceFromTarget = 10;
+                    Point helpTextLocation = new Point();
+                    Point arrowSourceLocation = new Point();
+                    Point arrowTargetLocation = new Point();
+                    //focused item top left corner
+                    if (controlToFocusLocation.X >= textNeededWidth && controlToFocusLocation.Y >= textNeededHeight)
+                    {
+                        helpTextLocation.X = controlToFocusLocation.X - textNeededWidth - arrowNeededLength; ;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this empty statement.](https://app.codacy.com/app/Ginger-Automation/Ginger/pullRequest?prid=4555652)